### DISCUSSION TITLE
Can't access matches array as array

### DISCUSSION THREAD

**Discussion by jayadehart** [react-hook-form/react-hook-form#5588] [Q&A]
I am using yup with yupresolver to create a schema for my form. That schema contains multiple matches each with their own message. 
  
          const schema = yup.object().shape({
            email: yup.string().email().required(),
            password: yup
              .string()
              .required("You must enter a password")
              .min(8, "Be at least 8 characters")
              .matches(/.*[0-9].*/, { message: "At least one number" })
              .matches(/^(?=[^A-Za-z]*[A-Za-z])[\x00-\x7F]*$/, {
                message: "At least one letter",
              })
              .matches(/^.*(?=.*[A-Z]).*$/, { message: "At least one capital" }),
          });

I want to render these messages to their own element. However, when I try to access the matches array at `errors.password.types?.matches`, is telling me I cannot use array methods like map on this value to render each error to its own element.  However I can clearly see in the errors object that it should be an array 

![however its clearly an array](https://user-images.githubusercontent.com/71462092/121789033-498e0600-cb87-11eb-96b7-2a33cff1bd37.png)

`{errors.password.types?.matches.map((item)=>(<p>{item}</p>))}`

When I just output `{errors.password.types?.matches}' it gives me a string with all the errors in the array concatenated. Looks like this:
![what it looks like](https://user-images.githubusercontent.com/71462092/121789039-5b6fa900-cb87-11eb-9e46-460aa36e08d9.png)

Please help this is incredibly frustrating

**Comment by bluebill1049** [comment]
<img width="870" alt="Screen Shot 2021-06-14 at 7 10 37 pm" src="https://user-images.githubusercontent.com/10513364/121868340-35dbce80-cd44-11eb-8f6a-6ab4d17f494c.png">

have you set `criteriaMode` to all? so it will return all the error messages?

**Comment by jayadehart** [comment]
I have. I think the issue is that sometimes matches returns an array, but
if there is only one item in it it returns a string. This makes it hard to
use array methods on it. I think it might be better if it always returned
an array regardless of the number of errors. Anyways thanks for reaching
out.

Jaya
